src/App.tsx

```diff
@@ -1,24 +1,196 @@
+
+  const hasAddTodoErrorTimerId = useRef(-1);
+  const hasGetTodoErrorTimerId = useRef(-1);
+  const hasDeleteTodoErrorTimerId = useRef(-1);
+  const hasUpdateTodoErrorTimerId = useRef(-1);
+
+  const inputFieldRef = useRef<HTMLInputElement | null>(null);
+
+  const hasTodos = todos.length > 0;
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos(USER_ID)
+
+      .then((data) => {
+        setIsLoading(false);
+        setTodos(data);
+
+        setErrorMessage('');
+      })
+
+      .catch(() => {
+        setIsLoading(false);
+        setErrorMessage('Unable to load todos');
+      });
+
+    hasGetTodoErrorTimerId.current = waitToClose(3000, setErrorMessage);
+  }, []);
+
+  const filteredTodo = useMemo(() => todos.filter(todo => {
+    switch (selectedStatus) {
+      case TodoStatus.Active:
+        return !todo.completed;
+
+      case TodoStatus.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  }), [selectedStatus, todos]);
+
+  const countOfNotCompletedTodo = useMemo(() => todos
+    .filter((todo) => !todo.completed).length, [todos]);
+
+  const allTodoCompleted = useMemo(() => filteredTodo
+    .every(({ completed }) => completed), [filteredTodo]);
+
+  const atleastOneTodoCompleted = useMemo(() => filteredTodo
+    .some(({ completed }) => completed), [filteredTodo]);
+
+  const handleDeleteCompleted = async () => {
+    setIsLoading(true);
+
+    const requests = todos
+      .filter((todo) => todo.completed)
+      .map(async (todo) => {
+        setDeletedTodoIds(prevTodos => [...prevTodos, todo.id]);
+
+        await deleteTodo(todo.id);
+        setTodos(prev => prev.filter(({ id }) => todo.id !== id));
+      });
+
+    try {
+      await Promise.all(requests);
+
+      setIsLoading(false);
+      setErrorMessage('');
+
+      setTodos(prev => prev.filter(todo => !todo.completed));
+    } catch (errorUpdate) {
+      setIsLoading(false);
+
+      setErrorMessage('Unable to delete a todo');
+
+      setDeletedTodoIds(() => []);
+      // eslint-disable-next-line no-console
+      console.warn(errorUpdate);
+
+      hasDeleteTodoErrorTimerId.current = waitToClose(3000, setErrorMessage);
+    }
+  };
+
+  const handleToggleButton = async () => {
+    setIsLoading(true);
+
+    const requests = todos
+      .map(async (todo) => {
+        setDeletedTodoIds(prevTodos => [...prevTodos, todo.id]);
+
+        await updateTodo(todo.id, {
+          ...todo,
+          completed: !allTodoCompleted,
+        });
+      });
+
+    try {
+      await Promise.all(requests);
+
+      setTodos(prevTodos => prevTodos.map(prevTodo => ({
+        ...prevTodo,
+        completed: !allTodoCompleted,
+      })));
+
+      setDeletedTodoIds(() => []);
+
+      setErrorMessage('');
+      setIsLoading(false);
+    } catch (error) {
+      setErrorMessage('Unable to update a todo');
+
+      setIsLoading(false);
+      setDeletedTodoIds(() => []);
```

Comment:
```suggestion
      setDeletedTodoIds([]);
```
If you don't need the previous state you don't need functional `setState`